packages/react-components/react-datepicker-compat/src/components/DatePicker/useDatePicker.tsx

```diff
@@ -204,8 +203,14 @@ export const useDatePicker_unstable = (props: DatePickerProps, ref: React.Ref<HT
     (newState: boolean) => {
       onOpenChange?.(newState);
       setOpenState(newState);
+
+      if (!open && !props.disabled) {
+        focus();
+      }
     },
-    [onOpenChange, setOpenState],
+    // Focus function keeps changing, so we need to skip it in the deps
```

Comment:
I think you should stabilize `focus()` rather than omit it here. The lint rule doesn't provide a mechanism to ignore only `focus` so this has the potential to allow bugs in the future.

Looking at the implementation of `useFocusLogic()` I think you could do something as simple as:

```ts
const useFocusLogic = () => {
   return React.useMemo(() => {
       // the previous useFocusLogic function body here
   }, []);
};
```

Or perhaps make the `focus` and `preventNextFocusingOpenPicker` functions refs as that will give you stable references as well.

Reply:
I think using React.useCallback for this should be good.